docker/kopia-build/Dockerfile

```diff
@@ -0,0 +1,42 @@
+FROM golang:1.17-alpine3.14 AS builder
+
+ARG kopiaBuildCommit
+ARG kopiaRepoOrg
+
+RUN apk add git
+
+WORKDIR /
+
+RUN git clone https://github.com/${kopiaRepoOrg}/kopia.git
+WORKDIR /kopia
+
+RUN git checkout ${kopiaBuildCommit}
+
+# Build static executable
+RUN GO111MODULE=on CGO_ENABLED=0 GO_EXTLINK_ENABLED=0 go build \
+  -ldflags="-s -w -X github.com/kopia/kopia/repo.BuildVersion=${kopiaBuildCommit}" \
```

Comment:
@ankitjain235 since kopia users goreleaser, can't we use goreleaser command to build the binary?

Reply:
There also we would have to build and push the docker image only right? Since we are only consuming the kopia binary in kanister-tools, we can't publish it as an artifact as well, and just cloning and building the binary from a specific commit seems better through a Dockerfile instead of a script that would do that.

Reply:
You mean goreleaser config in Kopia doesn't create a binary?

Reply:
It does, but for building the binary from kanister, we would have to clone kopia before building, and that won't be possible unless we do this through a script right?

Reply:
I did not get that.
So in this dockerfile, we are cloning kopia github repo and executing `go build` to generate binary. My question is, instead of `go build` command, can't we use `goreleaser` in kopia to build kopia binary?

Reply:
that will reduce maintenance efforts to keep the build command used in goreleaser config in sync with the one we are using in Dockerfile.